### PR TITLE
Upgrade to React 15.2.1 [try 3]

### PR DESCRIPTION
**Pull request by benvinegar**
cc @getsentry/team @fotinakis

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3747)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -110,7 +110,7 @@ test-acceptance:
 	@echo "--> Building static assets"
 	@${NPM_ROOT}/.bin/webpack
 	@echo "--> Running acceptance tests"
-	py.test tests/acceptance || exit 1
+	py.test tests/acceptance/test_emails.py::EmailTestCase::test_assigned_txt || exit 1
```
Comment (benvinegar) [inline Makefile]: At this point we're running a single test that doesn't even render HTML.

**Comment by benvinegar** [comment]
Still blocked on acceptance tests/Selenium failures:

```bash
py.test tests/acceptance/test_emails.py::EmailTestCase::test_assigned_txt || exit 1
21:36:16 [INFO] sentry.utils.raven.SentryInternalClient: Raven is not configured (logging is disabled). Please see the documentation for more information.
============================= test session starts ==============================
platform linux2 -- Python 2.7.9 -- py-1.4.26 -- pytest-2.6.4
plugins: django, cov, html, timeout, xdist
collected 23 items 
tests/acceptance/test_emails.py F
=================================== FAILURES ===================================
_______________________ EmailTestCase.test_assigned_txt ________________________
Traceback (most recent call last):
  File "/home/travis/build/getsentry/sentry/tests/acceptance/test_emails.py", line 12, in setUp
    self.login_as(self.user)
  File "/home/travis/build/getsentry/sentry/src/sentry/testutils/cases.py", line 111, in login_as
    self.save_session()
  File "/home/travis/build/getsentry/sentry/src/sentry/testutils/cases.py", line 448, in save_session
    value=self.session.session_key,
  File "/home/travis/build/getsentry/sentry/src/sentry/utils/pytest.py", line 97, in save_cookie
    self.get('/')
  File "/home/travis/build/getsentry/sentry/src/sentry/utils/pytest.py", line 37, in get
    self.driver.get(self.route(path), *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 248, in get
    self.execute(Command.GET, {'url': url})
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 234, in execute
    response = self.command_executor.execute(driver_command, params)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/selenium/webdriver/remote/remote_connection.py", line 401, in execute
    return self._request(command_info[0], url, body=data)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/selenium/webdriver/remote/remote_connection.py", line 469, in _request
    resp = opener.open(request, timeout=self._timeout)
  File "/opt/python/2.7.9/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/opt/python/2.7.9/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/opt/python/2.7.9/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/opt/python/2.7.9/lib/python2.7/urllib2.py", line 1227, in http_open
    return self.do_open(httplib.HTTPConnection, req)
  File "/opt/python/2.7.9/lib/python2.7/urllib2.py", line 1200, in do_open
    r = h.getresponse(buffering=True)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/raven/breadcrumbs.py", line 328, in getresponse
    rv = real_getresponse(self, *args, **kwargs)
  File "/opt/python/2.7.9/lib/python2.7/httplib.py", line 1073, in getresponse
    response.begin()
  File "/opt/python/2.7.9/lib/python2.7/httplib.py", line 415, in begin
    version, status, reason = self._read_status()
  File "/opt/python/2.7.9/lib/python2.7/httplib.py", line 379, in _read_status
    raise BadStatusLine(line)
BadStatusLine: ''
```

**Comment by fotinakis** [comment]
@benvinegar @dcramer , do you see these failures locally? Have you tried different browsers / webdrivers — like, switch from PhantomJs to the Chrome or Firefox Selenium driver and see if something weird is happening locally, like the browser is failing on some JS error or other error. I find Selenium environments often hide the real error because Selenium is terrible at raising up what is actually causing problems like this. Also, PhantomJs usually gives me no ends of trouble so most companies I see now run a real browser with a virtual framebuffer in CI (like xvfb) — if you try a normal browser locally and it works where PhantomJs doesn't, there are some Travis docs here about how to setup xvfb so you can run the same browser in Travis: https://docs.travis-ci.com/user/gui-and-headless-browsers/.

**Comment by benvinegar** [comment]
Nobody gets the failures locally.

@dcramer has opened up a PR to try using Firefox instead: https://github.com/getsentry/sentry/pull/3733/files. I experimented w/ this briefly locally but it didn't run properly for me (Firefox just opened then didn't do anything, but I think it works in CI). I'll try rebasing off of that next.

**Comment by benvinegar** [comment]
^Firefox patch is failing with same error (presuming it is running Firefox successfully).

**Comment by benvinegar** [comment]
At this point I suspect that the failures are from the Django server, seeing as neither browser can fetch any page successfully (BadStatusLine suggests the server is non-responsive / not even returning 500-level errors). Not clear to me how these package changes might impact that though ...

**Comment by dcramer** [comment]
The error (last I saw) is from selenium, not from the web server itself. It's not possible for us to test the response code (and other things) through selenium. 






On Mon, Jul 18, 2016 at 3:22 PM -0700, "Ben Vinegar" <notifications@github.com> wrote:












At this point I suspect that the failures are from the Django server, seeing as neither browser can fetch any page successfully (BadStatusLine suggests the server is non-responsive / not even returning 500-level errors). Not clear to me how these package changes might impact that though ...

**Comment by macqueen** [comment]
Reviewed 16 of 18 files at r1, 2 of 4 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3747)*

**Comment by benvinegar** [comment]
Reviewed 1 of 4 files at r5.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3747)*

**Pull request merged**